engine/language-server/src/main/scala/org/enso/languageserver/search/handler/InvalidateModulesIndexHandler.scala

```diff
@@ -34,22 +34,30 @@ final class InvalidateModulesIndexHandler(
 
   private def requestStage: Receive = {
     case SearchProtocol.InvalidateModulesIndex =>
-      runtime ! Api.Request(
+      val request = Api.Request(
         UUID.randomUUID(),
         Api.InvalidateModulesIndexRequest()
       )
+      runtime ! request
       val cancellable =
         context.system.scheduler.scheduleOnce(timeout, self, RequestTimeout)
-      context.become(responseStage(sender(), cancellable))
+      context.become(responseStage(sender(), request, cancellable))
   }
 
   private def responseStage(
     replyTo: ActorRef,
+    request: Api.Request,
     cancellable: Cancellable
   ): Receive = {
     case RequestTimeout =>
-      replyTo ! RequestTimeout
-      context.stop(self)
+      logger.warn(
+        "Failed to receive a [{}] response in [{}].",
+        request,
+        timeout
+      )
+      val newCancellable =
+        context.system.scheduler.scheduleOnce(timeout, self, RequestTimeout)
```

Comment:
Is the fix based on rescheduling `self` in case one gets a `RequestTimeout`?

Reply:
Yes and no. We now reschedule timeouts and not fail immediately. Rescheduling is to give more time for the command to finish and at the same time give us some information that things are slow.